.github/workflows/provider.yml

```diff
@@ -0,0 +1,168 @@
+name: Build and, optionally, release a capability provider
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        description: Provider name
+        required: true
+        type: string
+      capability:
+        description: Capability implemented by the provider
+        required: true
+        type: string
+    secrets:
+      subject:
+        description: Capability provider issuer subject key
+      issuer:
+        description: Capability provider issuer key
+      azurecr_username:
+        description: wasmCloud AzureCR username
+      azurecr_password:
+        description: wasmCloud AzureCR password
+
+jobs:
+  package:
+    runs-on: ubuntu-22.04-16-cores
+    steps:
+    - uses: actions/checkout@v4.1.1
+    - name: Extract tag context
+      id: ctx
+      run: |
+        sha_short=$(git rev-parse --short HEAD)
+        echo "owner=${GITHUB_REPOSITORY_OWNER,,}" >> "$GITHUB_OUTPUT"
+        echo "sha_short=${sha_short}" >> "$GITHUB_OUTPUT"
+
+        if [[ ${GITHUB_REF} == refs/tags/provider-${{ inputs.name }}-v* ]]; then
+          version=${GITHUB_REF_NAME#provider-${{ inputs.name }}-v}
+          echo "version=${version}" >> "$GITHUB_OUTPUT"
+          echo "tagged version is ${version}"
+          if [[ $version == *"-"* ]]; then
+            echo "tagged version ${version} is a pre-release"
+            echo "prerelease=true" >> "$GITHUB_OUTPUT"
+          fi
+        else
+          version=$(cargo metadata --manifest-path "./crates/provider-${{ inputs.name }}/Cargo.toml" --no-deps --format-version 1 | jq -r '.packages[] | select(.name == "wasmcloud-provider-${{ inputs.name }}") | .version')
+          echo "untagged version is ${version}"
+          echo "untagged version ${version} is a pre-release"
+          echo "version=${version}" >> "$GITHUB_OUTPUT"
+          echo "prerelease=true" >> "$GITHUB_OUTPUT"
+        fi
+
+    - uses: actions/download-artifact@v4
+      with:
+        path: artifacts
+
+    - run: chmod +x "./artifacts/wasmcloud-aarch64-apple-darwin/bin/${{ inputs.name }}"
+    - run: chmod +x "./artifacts/wasmcloud-aarch64-unknown-linux-musl/bin/${{ inputs.name }}"
+    - run: chmod +x "./artifacts/wasmcloud-x86_64-apple-darwin/bin/${{ inputs.name }}"
+    - run: chmod +x "./artifacts/wasmcloud-x86_64-pc-windows-msvc/bin/${{ inputs.name }}.exe"
+    - run: chmod +x "./artifacts/wasmcloud-x86_64-unknown-linux-musl/bin/${{ inputs.name }}"
+
+    - run: mv "./artifacts/wasmcloud-x86_64-unknown-linux-musl/bin/wash" wash
+    - run: chmod +x wash
+
+    - run: |
+        if [ "${{ secrets.issuer }}" != '' ]; then
```

Comment:
Nit: Might just be easier to have these be set as env variables for this step rather than exporting in the script

Reply:
That was my initial approach as well, however having these set to empty string breaks `wash` unfortunately, so the only way to support key autogeneration here is to check for the value explicitly before setting the env

Reply:
We wouldn’t want it to autogenerate keys though if they’re not set, right?

Reply:
That would prevent this from running on forks, meaning that this would never run on PRs from forks. I do think that we want to build PARs always, at least for the convenience of testing proposed changes, but also to ensure a PR does not break this process

Reply:
Yeah I can definitely get behind that